Objective-C/Tests/ReplicatorTest+PendingDocIds.m

```diff
@@ -129,18 +134,249 @@ - (void) testPendingDocIDsPullOnlyException {
+                                                     continuous: NO];
+    int i = arc4random_uniform(kNoOfDocument);
+    NSString* docId = [NSString stringWithFormat: kDocIdFormat, i];
+    config.pushFilter = ^BOOL(CBLDocument* document, CBLDocumentFlags flags) {
+        return [document.id isEqualToString: docId];
+    };
+
+    __block id<CBLListenerToken> token;
+    __block CBLReplicator* replicator;
+    __weak typeof(self) wSelf = self;
+    __block NSSet* docIds = [NSSet setWithObject: docId];
+    [self run: config reset: NO errorCode: 0 errorDomain: nil onReplicatorReady: ^(CBLReplicator* r) {
+        replicator = r;
+        
+        token = [replicator addChangeListener: ^(CBLReplicatorChange* change) {
+            __strong id strongSelf = wSelf;
+            
+            NSError* err = nil;
+            NSSet* ids = [change.replicator pendingDocumentIds: &err];
+            AssertNil(err);
+            
+            [strongSelf validateAction: kCreateActionValue forDocIds: docIds];
+            
+            if (change.status.activity == kCBLReplicatorConnecting) {
+                Assert([ids isEqualToSet: docIds]);
+                AssertEqual(ids.count, 1);
+            } else if (change.status.activity == kCBLReplicatorStopped) {
+                AssertEqual(ids.count, 0);
+            }
+        }];
+    }];
+    [replicator removeChangeListenerWithToken: token];
+}
+
+#pragma mark - IsDocumentPending API
+
+- (void) testIsDocumentPendingPullOnlyException {
+    id target = [[CBLDatabaseEndpoint alloc] initWithDatabase: otherDB];
+    id config = [self configWithTarget: target type: kCBLReplicatorTypePull continuous: NO];
+    
+    __block id<CBLListenerToken> token;
+    __block CBLReplicator* replicator;
+    __weak typeof(self) wSelf = self;
+    [self run: config reset: NO errorCode: 0 errorDomain: nil onReplicatorReady: ^(CBLReplicator* r) {
+        replicator = r;
+        
+        token = [replicator addChangeListener: ^(CBLReplicatorChange* change) {
+            if (change.status.activity == kCBLReplicatorBusy) {
+                [wSelf expectError: CBLErrorDomain code: CBLErrorUnsupported in: ^BOOL(NSError** err) {
+                    return [change.replicator isDocumentPending: @"document-id" error: err];
+                }];
+            }
+        }];
+    }];
+    
+    [replicator removeChangeListenerWithToken: token];
+}
+
+- (void) testIsDocumentPendingWithCreate {
+    NSString* docId = @"doc-1";
+    CBLMutableDocument* doc = [self createDocument: docId];
+    [doc setString: kCreateActionValue forKey: kActionKey];
+    [self saveDocument: doc];
+    
+    [self validateIsDocumentPending: docId isPresent: YES];
+    
+    docId = @"doc-2";
+    doc = [self createDocument: docId];
+    [doc setString: kCreateActionValue forKey: kActionKey];
+    [self saveDocument: doc];
+    [self validateIsDocumentPending: @"no-doc" isPresent: NO];
+}
+
+- (void) testIsDocumentPendingWithUpdate {
```

Comment:
What would you think of adding the corresponding text from https://docs.google.com/spreadsheets/d/1VJ8wjUw9loO_jcJYRAq3f2JJvbI4OiSdtpX6uwCpdxw/edit#gid=0 to the head of each of these tests?  The thing is, when one of these tests fails, it is going to be hard for some poor dev to figure out what the test is trying to test...

Reply:
It comes naturally to use create, update, delete and purge. I would vote for `update` than `edit`.

Reply:
updated the test sheet with the name